tests/test_dataset.py

```diff
@@ -311,6 +311,42 @@ def test_props_equal(self, constructor):
         assert list_props_uncached == list_props_cached
 
 
+@pytest.mark.parametrize(
+    "db",
+    [
+        "medium_db",
+    ],
+)
+def test_dataset_cache_root(tmpdir, request, db):
+    """Test configuration of cache root.
+
+    ``cache_root`` can be provided by different options,
+    in the following hierarchy:
+
+    * as argument to ``audbcards.Dataset()``
+    * as environment variable ``AUDBCARDS_CACHE_ROOT``
+    * as ``audbcards.config.CACHE_ROOT`` entry
+
+    """
+    db = request.getfixturevalue(db)
+    cache_root1 = audeer.mkdir(tmpdir, "cache1")
+    cache_root2 = audeer.mkdir(tmpdir, "cache2")
+    cache_root3 = audeer.mkdir(tmpdir, "cache3")
+    assert audbcards.config.CACHE_ROOT == "~/.cache/audbcards"
```

Comment:
Probably it is overkill to isolate the assertions into single tests.
What I think could come in handy would be a single lince comment for each of the assertions.